### ISSUE TITLE
Woo Email Confirmation - Add step for modifying email address

### ISSUE THREAD

**Issue by blackjackkent** [Automattic/wp-calypso#64164]
If the user selects "Edit Email Address" on the [new email confirmation screen](https://github.com/Automattic/wp-calypso/issues/64162), they should be directed to a new step where they can fill out a new email and their existing password. Submitting this form should trigger the same behavior that currently occurs when updating email address on Calypso account management.

<img width="929" alt="image" src="https://user-images.githubusercontent.com/13437011/171038268-07322b76-6675-4ec4-8597-f6fd6668cb85.png">

See DekZ88lDqG8Je04SyG1ZBK-fi-0%3A1 for detailed flow information.

**Comment by markbiek** [comment]
Noting that we need to `POST` the following to `/rest/v1.1/me/settings` to change the email address.

```
{
  user_email: "new@email.com"
}
```

Response:
```
{
  user_email: "current@email.com",
  user_email_change_pending: true,
  new_user_email: "new@email.com",
}
```

This automatically triggers sending a verification email. We don't need to send it separately.

Sending the following `POST` request to `/rest/v1.1/me/settings` will cancel the email change:
```
{
  user_email_change_pending: false,
}
```

**Comment by markbiek** [comment]
After some discussion, the consensus is that the password field isn't necessary.

Details in p1654701980575359-slack-C0Q664T29

**Comment by markbiek** [comment]
@vinimotaa @SaxonF My plan was to display something like this after the "Send verification" button is clicked

![image](https://user-images.githubusercontent.com/917632/172686037-7bdb46b0-9b76-45ae-8f11-b16d21d96352.png)

Let me know if you'd prefer a different approach.

**Comment by vinimotaa** [comment]
Hi @markbiek - thanks. Where would this message appear? Can you give a bit more context?

**Comment by markbiek** [comment]
Hi @vinimotaa!

There isn't a next step or place for the user to go once they click the "Send verification" button.

So my thought was to hide the "Send" button after it's been clicked and display the above message in its place.

**Comment by markbiek** [comment]
@vinimotaa One other note. I don't think we need the "Confirm my email later" link at the top on this screen.

They'll only be coming to this screen from
![image](https://user-images.githubusercontent.com/917632/172696206-1a5d0f33-0ee6-4c1e-ac12-10f456c3e519.png)
or
![image](https://user-images.githubusercontent.com/917632/172696239-0173e06a-5d23-4267-90ea-1079eac2f78d.png)

So they could conceivably go "Back" from this screen but it doesn't make sense to skip it since they aren't going anywhere from this point.

**Comment by vinimotaa** [comment]
Hey @markbiek - not trying to make things harder, but I put some thought into this; "My plan was to display something like this after the "Send verification" button is clicked".

I liked the suggestion of having a simple alert message - but I'm thinking of a case where a user doesn't receive the email once they are at the "edit email" screen. For that ideally, we should give them the option to resend the verification email, and even perhaps give them the chance to edit their email again in case they entered it wrongly.

You can find a simple flow for this here:
https://www.figma.com/file/HOg65lHD0QOfaq0QtzKQbC/Dog-Fooding?node-id=43%3A1128

Would you be able to check if that makes sense?

This approach uses a similar path that we have within WooCommerce - if the user has landed there, without verifying their email, we give them the same options/interactions through this "pop up alert message";

<img width="1595" alt="image" src="https://user-images.githubusercontent.com/97929097/172737186-b28318d0-e655-48f4-8733-78332b4d751a.png">

any thoughts?

**Comment by markbiek** [comment]
@vinimotaa Thanks for the update! I like the idea of sending them to a separate screen a lot.

But I'm wondering if we could just send them to this existing step instead of making a new one:

![image](https://user-images.githubusercontent.com/917632/172865130-9fb12661-6dcc-4b45-982e-3f058f6a50e2.png)

**Comment by vinimotaa** [comment]
Hi @markbiek yes definitely!

**Comment by markbiek** [comment]
@vinimotaa Splendid, thanks again for all the help!

**Comment by vinimotaa** [comment]
no worries M., hit me up for anything!

**Comment by markbiek** [comment]
Fixed by #64462
